src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs

```diff
@@ -780,5 +803,42 @@ private void CheckExplicitImplementationAccessor(MethodSymbol? thisAccessor, Met
                 diagnostics.Add(ErrorCode.ERR_ExplicitPropertyAddingAccessor, thisAccessor.GetFirstLocation(), thisAccessor, explicitlyImplementedEvent);
             }
         }
+
+        internal bool IsPartial => (this.Modifiers & DeclarationModifiers.Partial) != 0;
+
+        /// <summary>
+        /// <see langword="true"/> if this symbol corresponds to a semi-colon body declaration.
+        /// <see langword="false"/> if this symbol corresponds to a declaration with custom <see langword="add"/> and <see langword="remove"/> accessors.
+        /// </summary>
+        protected abstract bool IsFieldLike { get; }
```

Comment:
Was there a reason that existing member `EventSymbol.HasAssociatedField` wasn't suitable?

Reply:
Yes, `SourceFieldLikeEventSymbol` doesn't have `AssociatedField` if it's partial (or abstract or extern). But every partial event is composed of `SourceFieldLikeEventSymbol` and `SourceCustomEventSymbol` and this property is used to distinguish these two parts.

Reply:
Ah I think I get it. There's a need for essentially checking which syntactic form was used, which is unrelated to whether event actually has a field. And this is preferred over `is SourceFieldLikeEventSymbol` or other such tests. Thanks!